pkg/filesystem/subvolume_test.go

```diff
@@ -59,3 +63,35 @@ func TestGetOmapVal(t *testing.T) {
 		})
 	}
 }
+
+func TestGetSubvolumeNameFromPath(t *testing.T) {
+
+	tests := []struct {
+		path string
+		name string
+	}{
+		{
+			path: "/volumes/csi/csi-vol-6a99b552-fdcc-441d-b1e6-a522a85a503d/5f4e4caa-f835-41ba-83c1-5bbd57f6aedf",
+			name: "csi-vol-6a99b552-fdcc-441d-b1e6-a522a85a503d",
+		},
+		{
+			path: "",
+			name: "",
+		},
+		{
+			path: "/volumes/csi-vol-6a99b552-fdcc-441d-b1e6-a522a85a503d/5f4e4caa-f835-41ba-83c1-5bbd57f6aedf",
+			name: "5f4e4caa-f835-41ba-83c1-5bbd57f6aedf",
+		},
+		{
+			path: "csi-vol-6a99b552-fdcc-441d-b1e6-a522a85a503d/5f4e4caa-f835-41ba-83c1-5bbd57f6aedf",
+			name: "",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.path, func(t *testing.T) {
+			if name := getSubvolumeNameFromPath(tt.path); name != tt.name {
```

Comment:
i think we dont need if check here, asser.Equal will fail if we don't get the expected value